src/encoder/encoderopus.cpp

```diff
@@ -0,0 +1,439 @@
+// encoderopus.cpp
+// Create on August 15th 2017 by Palakis
+
+#include <QByteArray>
+#include <QMapIterator>
+
+#include "encoder/encoderopussettings.h"
+#include "engine/sidechain/enginesidechain.h"
+#include "util/logger.h"
+
+#include "encoder/encoderopus.h"
+
+namespace {
+// From libjitsi's Opus encoder:
+// 1 byte TOC + maximum frame size (1275)
+// See https://tools.ietf.org/html/rfc6716#section-3.2
+const int kMaxOpusBufferSize = 1+1275;
+// Opus frame duration in milliseconds. Fixed to 60ms
+const int kOpusFrameMs = 60;
+
+const mixxx::Logger kLogger("EncoderOpus");
+}
+
+EncoderOpus::EncoderOpus(EncoderCallback* pCallback)
+    : m_bitrate(0),
+      m_bitrateMode(0),
+      m_channels(0),
+      m_samplerate(0),
+	  m_readRequired(0),
+      m_pCallback(pCallback),
+      m_pFifoBuffer(nullptr),
+      m_pFifoChunkBuffer(nullptr),
+      m_pOpus(nullptr),
+      m_header_write(false),
+      m_packetNumber(0),
+      m_granulePos(0) {
+    m_opusComments.insert("ENCODER", "mixxx/libopus");
+    m_pOpusDataBuffer = new unsigned char[kMaxOpusBufferSize]();
```

Comment:
The size is known at compile time, so this array does not need to be allcated dynamically on the heap. And even if you decide to do so I would recommend using either std::vector or QVector.